diff -up vice-3.2/src/arch/unix/x11/ui-resources.c.htmlview vice-3.2/src/arch/unix/x11/ui-resources.c
--- vice-3.2/src/arch/unix/x11/ui-resources.c.htmlview	2018-06-28 16:45:31.277140071 +0200
+++ vice-3.2/src/arch/unix/x11/ui-resources.c	2018-06-28 16:46:40.220627630 +0200
@@ -183,7 +183,7 @@ static int set_confirm_on_exit(int val,
 #ifdef MACOSX_SUPPORT
 #define HTML_BROWSER_COMMAND_DEFAULT "/usr/bin/open %s"
 #else
-#define HTML_BROWSER_COMMAND_DEFAULT "firefox %s"
+#define HTML_BROWSER_COMMAND_DEFAULT "xdg-open %s"
 #endif
 
 static const resource_string_t resources_string[] = {
