diff -up vice-2.4.28.sdl/src/arch/sdl/ssi2001-drv.c~ vice-2.4.28.sdl/src/arch/sdl/ssi2001-drv.c
--- vice-2.4.28.sdl/src/arch/sdl/ssi2001-drv.c~	2016-05-18 23:40:30.000000000 +0200
+++ vice-2.4.28.sdl/src/arch/sdl/ssi2001-drv.c	2016-06-27 19:11:06.260782212 +0200
@@ -30,6 +30,7 @@
 
 #ifdef UNIX_COMPILE
 #include "../unix/ssi2001-drv.c"
+#include "../unix/io-access.c"
 #endif
 
 #if defined(WIN32_COMPILE) && !defined(__XBOX__)
