diff -up vice-2.0/src/arch/unix/x11/ui-resources.c.bar vice-2.0/src/arch/unix/x11/ui-resources.c
--- vice-2.0/src/arch/unix/x11/ui-resources.c.bar	2008-07-07 20:23:10.000000000 +0200
+++ vice-2.0/src/arch/unix/x11/ui-resources.c	2008-08-01 10:24:23.000000000 +0200
@@ -112,7 +112,7 @@ static const resource_string_t resources
 #ifdef MACOSX_SUPPORT    
       "/usr/bin/open %s",
 #else
-      "firefox %s",
+      "xdg-open %s",
 #endif
       RES_EVENT_NO, NULL,
       &ui_resources.html_browser_command,
