diff -up vice-2.3.9/src/arch/unix/archdep.c.tmpnam vice-2.3.9/src/arch/unix/archdep.c
--- vice-2.3.9/src/arch/unix/archdep.c.tmpnam	2011-06-17 18:19:22.000000000 +0200
+++ vice-2.3.9/src/arch/unix/archdep.c	2011-07-17 14:00:38.256652134 +0200
@@ -454,40 +454,26 @@ char *archdep_quote_parameter(const char
 
 char *archdep_tmpnam(void)
 {
-#if defined(GP2X) || defined(WIZ)
-    static unsigned int tmp_string_counter = 0;
-    char tmp_string[32];
-
-    sprintf(tmp_string, "vice%d.tmp", tmp_string_counter++);
-    return lib_stralloc(tmp_string);
-#else
-#ifdef HAVE_MKSTEMP
-    char *tmp_name;
-    const char mkstemp_template[] = "/vice.XXXXXX";
-    int fd;
-    char *tmp;
-    char *final_name;
-
-    tmp_name = lib_malloc(ioutil_maxpathlen());
-    if ((tmp = getenv("TMPDIR")) != NULL) {
-        strncpy(tmp_name, tmp, ioutil_maxpathlen());
-        tmp_name[ioutil_maxpathlen() - sizeof(mkstemp_template)] = '\0';
-    } else {
-        strcpy(tmp_name, "/tmp");
-    }
-    strcat(tmp_name, mkstemp_template);
-    if ((fd = mkstemp(tmp_name)) < 0) {
-        tmp_name[0] = '\0';
-    } else {
-        close(fd);
-    }
-
-    final_name = lib_stralloc(tmp_name);
-    lib_free(tmp_name);
-    return final_name;
+    char *tmp, *name;
+    
+    name = util_concat(archdep_home_path(), "/.vice", NULL);
+    if (access(name, F_OK)) {
+        mkdir(name, 0700);
+    }
+    lib_free(name);
+    
+    name = util_concat(archdep_home_path(), "/.vice/tmp", NULL);
+    if (access(name, F_OK)) {
+        mkdir(name, 0700);
+    }
+    tmp = tempnam(name, "vice");
+    lib_free(name);
+#ifndef LIB_DEBUG
+    return tmp;
 #else
-    return lib_stralloc(tmpnam(NULL));
-#endif
+    name = lib_stralloc(tmp);
+    free(tmp);
+    return name;
 #endif
 }
 
