diff -up vice-2.2/src/arch/unix/archdep.h.datadir vice-2.2/src/arch/unix/archdep.h
--- vice-2.2/src/arch/unix/archdep.h.datadir	2009-10-21 18:46:46.000000000 +0200
+++ vice-2.2/src/arch/unix/archdep.h	2010-06-22 10:06:21.353162984 +0200
@@ -120,7 +120,7 @@ extern const char *archdep_home_path(voi
 extern const char *archdep_pref_path;
 
 /* Define the default system directory (where the ROMs are).  */
-#ifdef __NetBSD__
+#if 1
 #define LIBDIR PREFIX "/share/vice"
 #else
 #define LIBDIR PREFIX "/lib/vice"
diff -up vice-2.2/src/arch/sdl/archdep_unix.h~ vice-2.2/src/arch/sdl/archdep_unix.h
--- vice-2.2/src/arch/sdl/archdep_unix.h~	2009-09-21 19:36:09.000000000 +0200
+++ vice-2.2/src/arch/sdl/archdep_unix.h	2010-06-22 11:24:46.205880453 +0200
@@ -85,7 +85,7 @@ extern const char *archdep_home_path(voi
 extern const char *archdep_pref_path;
 
 /* Define the default system directory (where the ROMs are).  */
-#ifdef __NetBSD__
+#if 1
 #define LIBDIR PREFIX "/share/vice"
 #else
 #define LIBDIR PREFIX "/lib/vice"
diff -up vice-2.4.3/configure~ vice-2.4.3/configure
--- vice-2.4.3/configure~	2013-02-26 20:45:36.000000000 +0100
+++ vice-2.4.3/configure	2013-03-02 15:13:00.479391490 +0100
@@ -20700,7 +20700,7 @@ fi
     ;;
   linux*)
     if test x"$host_cpu" = "xx86_64"; then
-      VICEDIR="$PREFIX/lib64/vice"
+      VICEDIR="$PREFIX/share/vice"
        if false; then
   ALTERNATE_DOCDIR_TRUE=
   ALTERNATE_DOCDIR_FALSE='#'
@@ -20710,7 +20710,7 @@ else
 fi
 
     else
-      VICEDIR="$PREFIX/lib/vice"
+      VICEDIR="$PREFIX/share/vice"
        if false; then
   ALTERNATE_DOCDIR_TRUE=
   ALTERNATE_DOCDIR_FALSE='#'
